extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -139,8 +125,18 @@ AmpBaseCarousel['props'] = {
   'snapBy': {attr: 'snap-by', type: 'number', media: true},
   'snapAlign': {attr: 'snap-align', type: 'string', media: true},
   'visibleCount': {attr: 'visible-count', type: 'number', media: true},
+  'children': {
+    props: {
```

Comment:
This confuses me. What does `props` insides a prop definition do?

Reply:
The child or children array will have props that we can access. For this example, you can have:
```jsx
<amp-base-carousel>
  <img src="big1.jpg" data-thumbnail-src="small1.jpg" />
  <img src="big2.jpg" data-thumbnail-src="small2.jpg" />
  <img src="big3.jpg" data-thumbnail-src="small3.jpg" />
</amp-base-carousel>
```

And the Preact component will be able to do:
```js
const thumbnails = [];
toChildArray(children).forEach(child => thumbanils.push(<img src={child.props.thumbnailSrc} />)
```

Reply:
This change was done as part of #30867. Let's follow up separately if we want to change it.